pinecone/config/config.py

```diff
@@ -47,11 +52,28 @@ def build(
         if not host:
             raise PineconeConfigurationError("You haven't specified a host.")
 
+        return Config(api_key, host, proxy_url, proxy_headers, ssl_ca_certs, ssl_verify, additional_headers)
+    
+    @staticmethod
+    def build_openapi_config(
+        config: Config, openapi_config: Optional[OpenApiConfiguration] = None, **kwargs
+    ) -> OpenApiConfiguration:
         if openapi_config:
-            openapi_config = copy.deepcopy(openapi_config)
-            openapi_config.host = host
-            openapi_config.api_key = {"ApiKeyAuth": api_key}
-        else:
-            openapi_config = OpenApiConfigFactory.build(api_key=api_key, host=host)
+            openapi_config = OpenApiConfigFactory.copy(openapi_config=openapi_config, api_key=config.api_key, host=config.host)
+        elif openapi_config is None:
+            openapi_config = OpenApiConfigFactory.build(api_key=config.api_key, host=config.host)
 
-        return Config(api_key, host, openapi_config, additional_headers)
\ No newline at end of file
+        # Check if value passed before overriding any values present
+        # in the openapi_config. This means if the user has passed 
+        # an openapi_config object and a kwarg for the same setting,
+        # the kwarg will take precedence.
```

Comment:
I may be reading this incorrectly, but it doesn't look like we're doing anything with `kwargs` here or in `build`?

Is the comment meant to be referencing overriding `openapi_config` with `config`?

Reply:
The `config` object is built using kwarg values. It happens in a different spot, but that's where those values come from. Sorry it's not more clear. Things get confusing when trying to manage and merge these different sources of configuration, which is a big reason I want to deprecate passing `openapi_config`.